### PR TITLE
Add PlatOnFloor function for agents

### PR DESCRIPTION
**Pull request by JaafarRammal**
# Summary

Add PlatOnFloor function to tell agents 

Fix design to only return foodTaken, not tower states (so no more -1 returned)

Edit from Jason: `TakeFood()` now returns two values, a `food.FoodType` and an `error`. `food.FoodType` is how much food an agent actually managed to take, and if `err != nil` then something went wrong - either you have already eaten, the platform is not on your floor, or you entered a negative amount of food to take.

### CODE REVIEW

**Review by hussain2603 — APPROVED** [review]
lgtm thanks!

**Review by VFjr — DISMISSED** [review]
lgtm

**Review by SaraFFdez — COMMENTED** [review]
Had to take a very quick look but the changes to team 3s code LGTM

**Comment by JeetendraJoshi** [comment]
LGTM.

**Review by JeetendraJoshi — COMMENTED** [review]
LGTM.

**Review by JeetendraJoshi — APPROVED** [review]
LGTM

**Comment by tomeaton17** [comment]
This is good.

**Review by tomeaton17 — APPROVED** [review]
It's good. Changes don't affect our agents functionality.

File: pkg/agents/team7/agent1/agent.go [hunk 1]
```diff
@@ -144,7 +145,16 @@ func (a *CustomAgent7) Run() {
 
 		//has the agent seen the platform
 		if a.CurrPlatFood() != -1 && !a.seenPlatform {
-			a.foodEaten = a.TakeFood(foodtotake)
+			foodEaten, err := a.TakeFood(foodtotake)
```
Comment (moinbukhari) [inline pkg/agents/team7/agent1/agent.go:148]: should be `a.foodEaten` instead of `foodEaten`

Comment (jzzheng22) [reply pkg/agents/team7/agent1/agent.go:148]: Doesn't work unfortunately, I tried that before. I do set it later on

**Comment by moinbukhari** [comment]
ok looks good from my side

**Review by MattScottEEE — APPROVED** [review]

**Pull request merged**